docs/test_specification.yaml

```diff
@@ -0,0 +1,218 @@
+definitions: 
+  hello1: 
+    - A signed hello_word image
+  hello2:
+    - An upgrade image, signed and marked for upgrade 
+  good_key:
+    - A key generated with a cryptosystem set in the bootloader with a correct value
+  bad_key:
+    - A key generated with a cryptosystem set in the bootloader with an incorrect value
+  wrong_key:   
+    - A key generated with a cryptosystem not matching the algorithm set in the bootloader
+  bootloader_output:
+    - 'unable to find bootable image'
+  hello1_output:
+    - 'Hello World from hello1'
+  hello2_output:
+    - 'Hello World from hello2'
+test_suites:
+  test_rsa_variants:
+    description:
+      - Test suite that verifies that boodloader correctly handles images with RSA signatures (good and bad)
+    test_level: system
+    test_type: feature
+    test_group: rsa
+    implemented: False  
+    test_cases:
+      test_good_rsa:
+        description:
+          - Test a good image, with a good upgrade, using RSA signatures
+        test setup:
+          - clean build directories
+          - clean hexes 
+        test procedure:
+          - build bootloader with overlay-rsa.conf
+          - build and sign hello1 with the good RSA key
+          - build and sign hello2 with the good RSA key
+          - erase chip
+          - flash bootloader
+          - Verify output. Expected bootloader_output
+          - flash hello1 hex
+          - Verify output. Expected hello1_output
+          - flash hello2 hex
+          - Verify output. Expected hello2_output
+          - reset chip
+          - Verify output. Expected hello1_output
+        success_criteria:
+          - Procedure complete without errors
+          - Output correctly verified
+      test_wrong_rsa:
+        description:
+          - Test a good image, with a wrong-signature upgrade, using RSA signatures.
+        test setup:
+          - clean build directories
+          - clean hexes 
+        test procedure:
+          - build bootloader with overlay-rsa.conf
+          - build and sign hello1 with the good RSA key
+          - build and sign hello2 with the bad RSA key
+          - erase chip
+          - flash bootloader
+          - Verify output. Expected bootloader_output
+          - flash hello1 hex
+          - Verify output. Expected hello1_output
+          - flash hello2 hex
+          - Verify output. Expected hello1_output
```

Comment:
I think that here you may also see log messages about  deleting the bad image.

Reply:
^^ that only for revert while dual-XIP mode is in usage, not in a image swap mode.
I see that test description should be extended by information on mcuboot configuration used.